pagerduty/resource_pagerduty_extension_test.go

```diff
@@ -0,0 +1,181 @@
+package pagerduty
+
+import (
+	"fmt"
+	"log"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+	"github.com/heimweh/go-pagerduty/pagerduty"
+)
+
+func init() {
+	resource.AddTestSweepers("pagerduty_extension", &resource.Sweeper{
+		Name: "pagerduty_extension",
+		F:    testSweepExtension,
+	})
+}
+
+func testSweepExtension(region string) error {
+	config, err := sharedConfigForRegion(region)
+	if err != nil {
+		return err
+	}
+
+	client, err := config.Client()
+	if err != nil {
+		return err
+	}
+
+	resp, _, err := client.Extensions.List(&pagerduty.ListExtensionsOptions{})
+	if err != nil {
+		return err
+	}
+
+	for _, extension := range resp.Extensions {
+		if strings.HasPrefix(extension.Name, "test") || strings.HasPrefix(extension.Name, "tf-") {
+			log.Printf("Destroying extension %s (%s)", extension.Name, extension.ID)
+			if _, err := client.Extensions.Delete(extension.ID); err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+func TestAccPagerDutyExtension_Basic(t *testing.T) {
+	extension_name := fmt.Sprintf("tf-%s", acctest.RandString(5))
+	extension_name_updated := fmt.Sprintf("tf-%s", acctest.RandString(5))
+	name := fmt.Sprintf("tf-%s", acctest.RandString(5))
```

Comment:
Nit: We have some helpers if you're interested:

- https://github.com/hashicorp/terraform/blob/7c6072c2a00b54dd3a7d0a19436052b2e4b553ab/helper/resource/id.go#L1

```go
extension_name := resource.PrefixedUniqueId("tf")
```